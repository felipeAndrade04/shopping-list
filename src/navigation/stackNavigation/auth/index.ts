export * from './AuthStackNavigation';
export * from './AuthStackNavigation.types';
