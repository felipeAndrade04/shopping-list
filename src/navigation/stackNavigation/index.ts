export * from './auth';
export * from './MainStackNavigation';
