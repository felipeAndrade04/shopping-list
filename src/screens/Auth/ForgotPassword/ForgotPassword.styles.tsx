import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  align-items: center;
  background-color: ${({ theme }) => theme.colors.background};
  padding: 0 24px;
`;

export const LogoContainer = styled.View`
  margin-bottom: 24px;
`;
