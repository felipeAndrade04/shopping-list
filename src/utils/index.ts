export * from './delay';
export * from './date';
