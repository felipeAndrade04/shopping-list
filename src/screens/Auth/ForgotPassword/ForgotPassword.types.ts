import { AuthStackNavigationProps } from '@app/navigation/stackNavigation/auth';

export interface ForgotPasswordProps {
  navigation: AuthStackNavigationProps;
}
