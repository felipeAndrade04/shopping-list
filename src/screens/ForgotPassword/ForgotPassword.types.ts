import { AuthStackNavigationProps } from '@app/navigation/stackNavigation/auth';

export interface ForgotPasswordProps {
  navigation: AuthStackNavigationProps;
}

export interface ForgotPasswordFormData {
  email: string;
}
