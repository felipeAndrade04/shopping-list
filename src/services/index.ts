import { auth, db } from '@app/config';
import { FirebaseAuthentication } from './auth/firebase-authentication';
import { FirebaseShopping } from './shopping/firebase-shopping';

const services = {
  auth: new FirebaseAuthentication(auth),
  shopping: new FirebaseShopping(db),
};

export default services;
