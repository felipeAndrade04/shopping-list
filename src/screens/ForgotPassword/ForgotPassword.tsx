import { Button, Input, Message, Spacer } from '@app/components';
import { AuthStackNavigationProps, AuthStackParamList } from '@app/navigation/stackNavigation/auth';
import React, { useState } from 'react';
import { StatusBar } from 'react-native';
import * as S from './ForgotPassword.styles';
import { useAuth } from '@app/hooks';

interface Props {
  navigation: AuthStackNavigationProps;
}

export function ForgotPassword({ navigation }: Props) {
  const [email, setEmail] = useState('');
  const { forgotPassword, isLoading } = useAuth();

  function onSubmit() {
    forgotPassword(email);
    onNavigation('Login');
  }

  function onNavigation(name: keyof AuthStackParamList) {
    navigation.navigate(name);
  }

  return (
    <S.Container>
      <StatusBar barStyle={'dark-content'} />
      <Message
        title="Recuperação de senha"
        description="Não se preocupe! Acontece. Insira o endereço associado à sua conta"
        showImage={false}
      />

      <Spacer dimesion={60} />

      <Input value={email} onChangeText={setEmail} placeholder="Informe o email registrado" />
      <Spacer dimesion={12} />

      <Button onPress={onSubmit} marginTop={24} marginBottom={24}>
        {isLoading ? 'Carregando' : 'Enviar'}
      </Button>
    </S.Container>
  );
}
