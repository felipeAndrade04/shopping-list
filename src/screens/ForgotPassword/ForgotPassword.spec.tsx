import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { ForgotPassword } from './ForgotPassword';
import { AuthStackNavigationProps } from '@app/navigation/stackNavigation/auth';
import services from '@app/services';
import { wrapper } from '@app/utils';

describe('Forgot password page', () => {
  const navigation = {
    ...jest.requireMock('@app/navigation/stackNavigation/auth'),
    navigate: jest.fn(),
  } as AuthStackNavigationProps;

  it('Should render correctly', () => {
    const tree = render(<ForgotPassword navigation={navigation}></ForgotPassword>, {
      wrapper,
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should be required email', async () => {
    const { queryByText } = render(<ForgotPassword navigation={navigation}></ForgotPassword>, {
      wrapper,
    });

    const button = queryByText(/enviar/i);

    await waitFor(() => {
      fireEvent.press(button);
    });

    expect(queryByText(/campo obrigatório/i)).toBeTruthy();
  });

  it('Should be valid email', async () => {
    const { queryByText, queryByTestId } = render(
      <ForgotPassword navigation={navigation}></ForgotPassword>,
      {
        wrapper,
      }
    );

    const button = queryByText(/enviar/i);
    const inputEmail = queryByTestId(/input-email/i);

    await waitFor(() => {
      fireEvent.changeText(inputEmail, 'teste');
      fireEvent.press(button);
    });

    expect(queryByText(/informe um email válido/i)).toBeTruthy();
  });

  it('Should be do forgot password correctly', async () => {
    const email = 'teste@email.com';
    jest.spyOn(services.auth, 'forgotPassword').mockImplementation(() => {
      return new Promise((resolve) => {
        resolve();
      });
    });
    const { queryByText, queryByTestId } = render(
      <ForgotPassword navigation={navigation}></ForgotPassword>,
      {
        wrapper,
      }
    );

    const button = queryByText(/enviar/i);
    const inputEmail = queryByTestId(/input-email/i);

    await waitFor(() => {
      fireEvent.changeText(inputEmail, email);
      fireEvent.press(button);
    });

    expect(services.auth.forgotPassword).toBeCalledWith(email);
    expect(navigation.navigate).toBeCalledWith('Login');
  });

  it('Should failed forgot password', async () => {
    const email = 'teste@email.com';
    jest.spyOn(services.auth, 'forgotPassword').mockImplementation(() => {
      return new Promise((_resolve, reject) => {
        return reject(new Error('Forgot password failed'));
      });
    });
    const { queryByText, queryByTestId } = render(
      <ForgotPassword navigation={navigation}></ForgotPassword>,
      {
        wrapper,
      }
    );

    const button = queryByText(/enviar/i);
    const inputEmail = queryByTestId(/input-email/i);

    await waitFor(() => {
      fireEvent.changeText(inputEmail, email);
      fireEvent.press(button);
    });

    expect(services.auth.forgotPassword).toBeCalled();
    expect(services.auth.forgotPassword).rejects.toThrow();
  });
});
