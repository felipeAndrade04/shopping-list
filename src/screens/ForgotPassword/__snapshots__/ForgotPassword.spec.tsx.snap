// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot password page Should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#EEEEEE",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 0,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 24,
        },
      ]
    }
  >
    <logo.svg />
  </View>
  <View
    style={
      Array [
        Object {
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#0D0A0B",
            "fontFamily": "Poppins-Medium",
            "fontSize": 24,
          },
        ]
      }
    >
      Recuperação de senha
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#0D0A0B",
            "fontFamily": "Poppins-Regular",
            "fontSize": 16,
            "marginTop": 14,
          },
        ]
      }
    >
      Não se preocupe! Acontece. Insira o endereço associado à sua conta
    </Text>
  </View>
  <View
    dimesion={60}
    style={
      Array [
        Object {
          "height": 60,
          "width": 60,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      isFocused={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#FFFFFF",
            "borderRadius": 6,
            "borderStyle": "solid",
            "borderWidth": 2,
            "flexDirection": "row",
            "opacity": 1,
            "transition": "border-color 0.3s ease-in",
            "width": "100%",
          },
        ]
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Informe o email registrado"
        style={
          Array [
            Object {
              "color": "#0D0A0B",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontSize": 16,
              "height": 50,
              "paddingBottom": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 8,
            },
          ]
        }
        testID="input-email"
        value=""
      />
    </View>
  </View>
  <View
    dimesion={12}
    style={
      Array [
        Object {
          "height": 12,
          "width": 12,
        },
      ]
    }
  />
  <RNGestureHandlerButton
    activeOpacity={0.105}
    collapsable={false}
    color="#009Cf7"
    delayLongPress={600}
    enabled={false}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    isLoading={false}
    marginBottom={24}
    marginTop={24}
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#009Cf7",
        "borderColor": "black",
        "borderRadius": 6,
        "borderStyle": "solid",
        "borderWidth": 0,
        "display": "flex",
        "height": 56,
        "justifyContent": "center",
        "marginBottom": 24,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 24,
        "opacity": 0.6,
        "paddingBottom": 8,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 8,
        "width": "100%",
      }
    }
    underlayColor="black"
    variant="solid"
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "black",
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderRadius": 6,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Text
      color="#009Cf7"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Poppins-Bold",
            "fontSize": 18,
          },
        ]
      }
      variant="solid"
    >
      Enviar
    </Text>
  </RNGestureHandlerButton>
</View>
`;
